CMakeLists.txt

```diff
@@ -19,15 +56,41 @@ if(NOT CMAKE_BUILD_TYPE)
     set (CMAKE_BUILD_TYPE  "RelWithDebInfo" CACHE STRING "Choose the type of build, options are: None(CMAKE_CXX_FLAGS or CMAKE_C_FLAGS used) Debug Release RelWithDebInfo MinSizeRel." FORCE)
 endif(NOT CMAKE_BUILD_TYPE)
 
-include_directories (inc)
-include_directories (lib)
+
+# Find packages to be linked
+find_package (Boost COMPONENTS filesystem iostreams program_options regex log thread system REQUIRED )
+include_directories (${Boost_INCLUDE_DIR} )
+find_package (Threads REQUIRED)
+set (EXT_LIB_PATH "/usr/local/lib" CACHE STRING "Absolute path to libraries installation folder.")
+
+
 
 # Application modules
+add_library(mod_sqlite3 STATIC lib/sqlite/sqlite3.c)
+
+set (THIRD_PARTY_LIBRARIES
```

Comment:
Writing it as `if (MAC) else` here would make things more clear and less redundant, because right now the libraries are set twice in MacOS.